model/task/task.go

```diff
@@ -2920,31 +2920,31 @@ func GetTasksByVersion(versionID string, opts GetTasksByVersionOptions) ([]Task,
 			},
 		})
 	}
-	countPipeline := []bson.M{}
-	countPipeline = append(countPipeline, pipeline...)
-	countPipeline = append(countPipeline, bson.M{"$count": "count"})
+
+	sortAndPaginatePipeline := []bson.M{}
 
 	sortFields := bson.D{}
 	if len(opts.Sorts) > 0 {
 		for _, singleSort := range opts.Sorts {
 			if singleSort.Key == DisplayStatusKey || singleSort.Key == BaseTaskStatusKey {
-				pipeline = append(pipeline, addStatusColorSort((singleSort.Key)))
+				sortAndPaginatePipeline = append(sortAndPaginatePipeline, addStatusColorSort((singleSort.Key)))
 				sortFields = append(sortFields, bson.E{Key: "__" + singleSort.Key, Value: singleSort.Order})
 			} else {
 				sortFields = append(sortFields, bson.E{Key: singleSort.Key, Value: singleSort.Order})
 			}
 		}
+	} else {
+		sortFields = append(sortFields, bson.E{Key: IdKey, Value: 1})
```

Comment:
Why do we now only do this in the else case?

Reply:
it seemed wrong, It would always apply a sort on the task id. Even though there were other sorts applied.  What do you think?

Reply:
I think it does seem reasonable to sort alphabetically after other sorts have been applied (like if we sort by status, sort task ID alphabetically within that, right?)

Reply:
the IdKey helps stabilize the sort if the sort condition is a tie. including this conditionally could cause our gql tests to sometimes fail https://docs.mongodb.com/manual/reference/operator/aggregation/sort/#sort-consistency